test/mainnet-fork/Uniswapv4.t.sol

```diff
@@ -0,0 +1,1971 @@
+        mainnetController.setUniswapV4TickLimits(bytes32(0), 0, 0, 0);
+    }
+
+    function test_setUniswapV4tickLimits_revertsForNonRelayer() external {
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IAccessControl.AccessControlUnauthorizedAccount.selector,
+                _unauthorized,
+                mainnetController.DEFAULT_ADMIN_ROLE()
+            )
+        );
+
+        vm.prank(_unauthorized);
+        mainnetController.setUniswapV4TickLimits(bytes32(0), 0, 0, 0);
+    }
+
+    function test_setUniswapV4tickLimits_revertsWhenInvalidTicks() external {
+        vm.prank(SPARK_PROXY);
+        vm.expectRevert("MC/invalid-ticks");
+        mainnetController.setUniswapV4TickLimits(bytes32(0), 1, 1, 1);
+
+        vm.prank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(bytes32(0), 0, 1, 1); // lower must be less than upper
+
+        vm.prank(SPARK_PROXY);
+        vm.expectRevert("MC/invalid-ticks");
+        mainnetController.setUniswapV4TickLimits(bytes32(0), 0, 1, 0);
+
+        vm.prank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(bytes32(0), 0, 0, 0); // maxTickSpacing can only be 0 if all 0
+    }
+
+    function test_setUniswapV4tickLimits() external {
+        vm.expectEmit(address(mainnetController));
+        emit MainnetController.UniswapV4TickLimitsSet(_POOL_ID, -60, 60, 20);
+
+        vm.record();
+
+        vm.prank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -60, 60, 20);
+
+        _assertReentrancyGuardWrittenToTwice();
+
+        ( int24 tickLowerMin, int24 tickUpperMax, uint24 maxTickSpacing ) = mainnetController.uniswapV4TickLimits(_POOL_ID);
+
+        assertEq(tickLowerMin,   -60);
+        assertEq(tickUpperMax,   60);
+        assertEq(maxTickSpacing, 20);
+    }
+
+    /**********************************************************************************************/
+    /*** Story Tests                                                                            ***/
+    /**********************************************************************************************/
+
+    /**
+     * @dev Story 1 is a round trip of liquidity minting, increase, decreasing, and closing/burning,
+     *      each 90 days apart, while an external account swaps tokens in and out of the pool.
+     *      - The relayer mints a position with 1,000,000 liquidity.
+     *      - The relayer increases the liquidity position by 50% (to 1,500,000 liquidity).
+     *      - The relayer decreases the liquidity position by 50% (to 750,000 liquidity).
+     *      - The relayer decreases the remaining liquidity position (to 0 liquidity).
+     */
+    function test_uniswapV4_story1() external {
+        // Setup the pool and the controller.
+        vm.startPrank(SPARK_PROXY);
+        mainnetController.setUniswapV4TickLimits(_POOL_ID, -60, 60, 20);
+        mainnetController.setMaxSlippage(address(uint160(uint256(_POOL_ID))), 0.98e18);
```

Comment:
Remove all maxSlippage setters in all testing

Reply:
Was the only one.